crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -49,8 +49,20 @@ pub struct Mesh {
 /// # use bevy_render::render_resource::PrimitiveTopology;
 /// fn create_triangle() -> Mesh {
 ///     let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
-///     mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, vec![[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [1.0, 1.0, 0.0]]);
-///     mesh.set_indices(Some(Indices::U32(vec![0,1,2])));
+///     mesh.insert_attribute(
+///         Mesh::ATTRIBUTE_POSITION,
+///         vec![[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [1.0, 1.0, 0.0], [0.0, 0.0, 0.0]]
+///     );
+///     // We have created 4 vertices, each with it's own position attribute.
+///     mesh.insert_attribute(
+///         Mesh::ATTRIBUTE_UV_0,
+///         vec![[0.0, 0.0], [1.0, 0.0], [0.0, 1.0], [1.0, 1.0]]
+///     );
+///     // We have assigned a UV coordinate for each of the 4 vertices we created.
+///     // NOTE: Multiple vertices can have the same attribute, but multiple attributes cannot be
+///     // of the same vertex.
+///     // We define the triangles using the defined vertices' indices.
+///     mesh.set_indices(Some(Indices::U32(vec![0,1,2 , 0,1,3])));
```

Comment:
Triangles have what is called a vertex winding order. By default in bevy we used right-handed coordinate conventions which means our default winding order is counter-clockwise. The indices 0,1,2 go from 1,0 to 0,1 to 1,1 which is clockwise. 0,1,3 goes from 1,0 to 0,1 to 0,0 which is counter-clockwise. This issue with 0,1,2 was present in the documentation before but it would be helpful to correct it and to note the default winding order. The winding order is used to understand whether the triangle is being viewed from the front or back. If it is viewed from the back then by default it will be culled and will not be visible.

Reply:
To clarify what's happening, because it's easier with visuals, here's your mesh:
![image](https://github.com/bevyengine/bevy/assets/134181069/aa9079e7-15d0-4b47-86b7-a9d70bb23536)
By "default", triangles are viewed only when their vertexes appear counter-clockwise, so the "purple"/first triangle (the one which was already there in the original example) would be viewed from the "back" only (with a camera at -Z), and the orange one viewed from the "front" only (with a camera at +Z).

Here is how your mesh looks from the front (I added the axis on the picture):
![image](https://github.com/bevyengine/bevy/assets/134181069/3801ebda-79f4-4271-a669-e5cadf5a6338)

and from the back:
![image](https://github.com/bevyengine/bevy/assets/134181069/6fd751e3-2e1a-44cc-a5fe-824812f58f6c)

Usually, when you have a flat mesh you would it to be viewed from the front by default (triangles face +Z or +Y with counter-clockwise vertices).

The default behavior is with `StandardMaterial.cull_mode = Some(Face::Front)`. `Some(Face::Back)` makes only the triangles with clockwise vertices be viewed, and `None` all the triangles no matter their vertex winding order.

I think those informations should be in the doc, can you try to put them in?

Also maybe have an image (is it possible? :x) with the mesh you generated rendered with the bevy icon as a texture, and an overlay of the vertex indices and the axis (like the first image in this comment), would be great to immediately visualise what the example do.